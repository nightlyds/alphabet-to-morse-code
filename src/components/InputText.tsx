/* eslint-disable react/require-default-props */
import React from "react";
import Translator from "../functions/Translator";

export type InputTextProps = {
    changeText?: (value: string) => void;
    firstLanguage?: string;
    secondLanguage?: string;
};

const InputText = ({
    changeText = () => {},
    firstLanguage = "ENG",
    secondLanguage = "MORSE",
}: InputTextProps) => {
    const textAreaChange = (
        event: React.ChangeEvent<HTMLTextAreaElement>
    ): void => {
        const {
            target: { value },
        } = event;

        const valueInString = `${value}`;

        changeText(Translator(firstLanguage, secondLanguage, valueInString));
    };
    return (
        <div className="input-text">
            <textarea
                className="input-text-textarea"
                onChange={e => textAreaChange(e)}
                placeholder="Type your text here.."
            />
        </div>
    );
};

export default InputText;
