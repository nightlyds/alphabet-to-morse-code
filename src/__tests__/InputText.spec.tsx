import React from "react";
import { shallow } from "enzyme";
import InputText from "../components/InputText";

let setUp = () => shallow(<InputText />);

describe("InputText tests", () => {
    describe("Snapshot tests", () => {
        let component: any;

        beforeEach(() => {
            component = setUp();
        });

        it("Basic snapshot test", () => {
            expect(component).toMatchSnapshot();
        });
    });
});
