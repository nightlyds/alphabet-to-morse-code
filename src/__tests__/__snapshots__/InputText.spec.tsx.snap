// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText tests Snapshot tests Basic snapshot test 1`] = `
<div
  className="input-text"
>
  <textarea
    className="input-text-textarea"
    onChange={[Function]}
    placeholder="Type your text here.."
  />
</div>
`;
